### ISSUE TITLE
Undecorated classes with DI migration not working TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string

### ISSUE THREAD

**Issue by elvisbegovic** [angular/angular#33143]
# 🐞 bug report

### Affected Package
The issue is caused by package [@angular/core/schematics/migrations/undecorated-classes-with-di/index.ts#L174](https://github.com/angular/angular/blob/master/packages/core/schematics/migrations/undecorated-classes-with-di/index.ts#L174)

### Is this a regression?

no, this migration code appear in v9 only not in v8

### Description

**Undecorated classes with DI migration** not working

## 🔬 Minimal Reproduction
1. ng update @angular/core --from 8.0.0 --to 9.0.0 --migrate-only"

## 🔥 Exception or Error
<pre><code>
... all previous migrations ARE OK
** Executing migrations for version 9.0.0-beta of package '@angular/core' **
    ------ Undecorated classes with DI migration ------

    This migration uses the Angular compiler internally and therefore projects that no longer build successfully after the update cannot run the migration. Please ensure there are no AOT compilation errors and
 rerun the migration.. The following project failed: tsconfig.app.json
    
    TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    
    Migration can be rerun with: "ng update @angular/core --from 8.0.0 --to 9.0.0 --migrate-only"
    Successfully migrated all found undecorated classes
    that use dependency injection.
    ----------------------------------------------

</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 9.0.0-next.9
Node: 12.9.1
OS: win32 x64
Angular: 9.0.0-next.10
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router, service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-next.9
@angular-devkit/build-angular     0.900.0-next.9
@angular-devkit/build-optimizer   0.900.0-next.9
@angular-devkit/build-webpack     0.900.0-next.9
@angular-devkit/core              9.0.0-next.9
@angular-devkit/schematics        9.0.0-next.9
@angular/cdk                      9.0.0-next.0
@angular/cli                      9.0.0-next.9
@angular/material                 9.0.0-next.0
@angular/pwa                      0.900.0-next.9
@ngtools/webpack                  9.0.0-next.9
@schematics/angular               9.0.0-next.9
@schematics/update                0.900.0-next.9
rxjs                              6.4.0
typescript                        3.5.3
webpack                           4.41.0
</code></pre>

**Anything else relevant?**
[tsconfig.app.json](https://gist.github.com/elvisbegovic/e215b63c06086a7958cc8a54f97aebd9)
[tsconfig.json](https://gist.github.com/elvisbegovic/91c0d5319c00e28c89b4003c5289aa94)
[rudimentary angular.json](https://gist.github.com/elvisbegovic/68b8ebb7741cbc57c3727d185bc09d12)
I do not have error on ng serve 

When debuging nodejs code `--inspect-brk` migration run without error (but im not sure it udpate my codebase)

**Comment by devversion** [comment]
I just saw this issue. Is this still relevant? We landed various fixes and this issue has been reported before v9 stable has been released.

I'm closing this in the meanwhile. Please feel free to create a new one, or respond here. Thanks!

**Comment by wittlock** [comment]
I've run into this issue trying to upgrade a project created on v8 and upgrading it to v9 running version 9.0.2 IIRC. It was stable for sure as I try to avoid pre-releases for production systems.

**Comment by devversion** [comment]
Interesting. Thanks for the quick reply. Do you think you would be able to re-run into this issue? If so, and you would be willing to help, that would be highly appreciated.

I'm specifically looking for simply re-running the update with the latest CLI beta version that includes https://github.com/angular/angular-cli/commit/66b218c5f6180bb62609d89ef691ef514fa359ec. That will give us a full stack trace of the error.

**Comment by devversion** [comment]
I'd be also happy to have simple reproduction steps. On my end I did not see this issue surface. I've tested various projects against this migration before v9 landed.

**Comment by wittlock** [comment]
Sure, I'll do another test run with the latest versions and see what I come up with. I'll see if I can figure out some reproduction example as well by stripping the project I'm trying to upgrade but no promises on that.

**Comment by wittlock** [comment]
Just tried migrating again using 9.0.4 and I see there's been some very nice updates to the migration process. It now pointed out the source file with an issue (needed an explicit constructor) and it even added a TODO-comment in the file for me. And then the migration concluded instead of failing and needing to be resumed.

So sometime between the version I tried before and 9.0.4 the error has been replaced with a very helpful message and comment which made resolving it trivial. So while issue was still around in some stable version it's absolutely resolved now for my case at least.

Thanks for the prod to try again with the latest fixes.

**Comment by devversion** [comment]
Thanks for trying the migration again! Very appreciated. I'm unsure though what change would have fixed this from stable v9 to the current patch release. I'll keep this closed until someone else jumps in and says that the issue is still present.

**Comment by StefanChristou** [comment]
I'm getting this error using ng update. Following the guide going from 8-9. Doesn't like my tsconfig.app.json 

`{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/app",
    "types": []
  },
  "files": [
    "main.ts",
    "polyfills.ts"
  ],
  "include": [
    "src/**/*.d.ts"
  ]
}
`

**Comment by devversion** [comment]
@StefanChristou Do you have a full error stack trace? I'm looking for more details: https://github.com/angular/angular/issues/33143#issuecomment-593891284.

**Comment by StefanChristou** [comment]
Error message for above comment is: 
`    This migration uses the Angular compiler internally and therefore projects that no longer build successfully after the update cannot run the migration. Please ensure there are no AOT compilation errors and rerun the migration. The following project failed: src/tsconfig.app.json
    
    TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    
    Could not migrate all undecorated classes that use dependency
    injection. Some project targets could not be analyzed due to
    TypeScript program failures.
    
    Migration can be rerun with: "ng update @angular/core --migrate-only migration-v9-undecorated-classes-with-di"
    
  Migration completed.
`

Tried re-running: "ng update @angular/core --migrate-only migration-v9-undecorated-classes-with-di" 

Looks like I'm missing a property from from src/tsconfig.app.json. But what I wonder??

**Comment by StefanChristou** [comment]
Full stack trace 

```➜  components git:(update/Angular-9) ng update @angular/core --migrate-only migration-v9-undecorated-classes-with-di
Using package manager: 'npm'
Collecting installed dependencies...
Found 88 dependencies.
** Executing 'migration-v9-undecorated-classes-with-di' of package '@angular/core' **

❯ Undecorated classes with DI migration.
  As of Angular 9, it is no longer supported to use Angular DI on a class that does not have an Angular decorator.
  Read more about this here: https://v9.angular.io/guide/migration-undecorated-classes
    
    This migration uses the Angular compiler internally and therefore projects that no longer build successfully after the update cannot run the migration. Please ensure there are no AOT compilation errors and rerun the migration. The following project failed: src/tsconfig.app.json
    
    TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    
    Could not migrate all undecorated classes that use dependency
    injection. Some project targets could not be analyzed due to
    TypeScript program failures.
    
    Migration can be rerun with: "ng update @angular/core --migrate-only migration-v9-undecorated-classes-with-di"
    
  Migration completed.


Your project has been updated to Angular version 9!
For more info, please see: https://v9.angular.io/guide/updating-to-version-9
```

**Comment by devversion** [comment]
Thanks @StefanChristou. Unfortunately the CLI doesn't print out the full stack trace, but just the error message. We added full-stack trace support in the latest next release of the CLI. 

When you get a chance, it would be great to re-run the update with the latest `@next` CLI version installed, or with `--next`. https://github.com/angular/angular-cli/commit/66b218c5f6180bb62609d89ef691ef514fa359ec. The full stack trace is then stored in a log file.

Let me know if that is possible. The current error doesn't help determining the root cause. I've spent time looking for potential `path` calls.
